flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/config/FlinkConfigBuilder.java

```diff
@@ -409,15 +419,44 @@ private static void setPodTemplate(
             effectiveConfig.setString(
                     podConfigOption,
                     createTempFile(
-                            mergePodTemplates(
-                                    basicPod,
-                                    appendPod,
-                                    effectiveConfig.get(
-                                            KubernetesOperatorConfigOptions
-                                                    .POD_TEMPLATE_MERGE_BY_NAME))));
+                            applyResourceToPodTemplate(
+                                    mergePodTemplates(
```

Comment:
I think we should not nest these calls but do one after the other for readability

Reply:
Updated